### PR TITLE
Add property to provide a http proxy url to connect to glue registry

### PR DESCRIPTION
**Pull request by abhilash47**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java
Comment (blacktooth) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java:101]: The indentation seems off, please fix it.

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java
Comment (blacktooth) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java:104]: This error message seems incorrect. Should this be `.getProxyUrl` instead? Also, the `String.format` parameters must have a placeholder `%s' for printing the Uri.

File: common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java
Comment (blacktooth) [inline common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java:21]: Indentation seems off.

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java
Comment (blacktooth) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java:102]: You might want to consider moving this to validation to `GlueSchemaRegistryConfiguration.java` similar to how it is being done for the configuration parameters.

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java:93]: @blacktooth  - Added unit tests

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java]: @blacktooth  moved the validation to GlueSchemaRegistryConfiguration.java

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java]: fixed the indentation

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java]: fixed it

File: common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java
Comment (abhilash47) [reply common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java]: fixed indentation

**Comment by abhilash47** [comment]
@blacktooth - I am not able to reproduce the failing windows builds on my windows laptop. Could you please help me with whats causing the failure ?

**Comment by blacktooth** [comment]
Thanks for checking! It looks like the Windows builds are failing irrespective of this change. We will look into those separately. This PR can be merged after approval.

**Comment by abhilash47** [comment]
@blacktooth - I have pulled the latest changes to resolve windows build issues and version upgrades. Could this PR be approved and merged ?

**Comment by YangSan0622** [comment]
@abhilash47 Hello, we are working on our internal process for this PR, we will update you once that process is completed.

**Comment by YangSan0622** [comment]
@abhilash47 Hello, we have got the approval needed for this PR, we will merge it shortly. Thanks a lot!

**Review by blacktooth — APPROVED** [review]

**Pull request merged**
